solutions/src/main/java/ru/sberbank/school/task02/FxConversionServiceImpl.java

```diff
@@ -0,0 +1,102 @@
+package ru.sberbank.school.task02;
+
+import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.Quote;
+import ru.sberbank.school.task02.util.Symbol;
+import ru.sberbank.school.task02.exception.FxConversionException;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+
+
+public class FxConversionServiceImpl implements FxConversionService {
+
+    ExternalQuotesService externalQuotesService;
+
+    FxConversionServiceImpl(ExternalQuotesService externalQuotesService) {
+        this.externalQuotesService = externalQuotesService;
+    }
+
+    @Override
+    public BigDecimal convert(ClientOperation operation, Symbol symbol, BigDecimal amount)
+            throws FxConversionException {
+
+        if (amount.compareTo(BigDecimal.valueOf(0)) < 0 ) {
+            throw new FxConversionException("Объем не может быть отрицательным");
+        }
+
+        List<Quote> quotes = new ArrayList<>();
+        quotes = externalQuotesService.getQuotes(symbol);
+
+        if (quotes.size() == 0) {
+            throw new FxConversionException("Список котировок ExternalQuotesService не должен быть пустой");
+        }
+
+        int negativeValueCounter = 0;
+        for (int i = 0; i < quotes.size(); i++) {
+            if (quotes.get(i).getVolumeSize().compareTo(BigDecimal.valueOf(0)) < 0) {
+                negativeValueCounter++;
+            }
+        }
+        if (negativeValueCounter == quotes.size()) {
+            throw new FxConversionException("Список котировок ExternalQuotesService содержит только отрицательные value");
+        }
+
+        //Создание списка котировок значение которых больше, чем amount
+        List<Quote> quotesBiggerThanAmount = new ArrayList<>();
+        for (int i = 0; i < quotes.size(); i++) {
+
+            if (quotes.get(i).getVolumeSize().compareTo(amount) > 0
+                    && quotes.get(i).getVolumeSize().compareTo(BigDecimal.valueOf(0)) > 0) {
+                quotesBiggerThanAmount.add(quotes.get(i));
+            }
+        }
+
+        //Если amount больше самого большого value, то вернуть котировку для минимального value больше 0 из quotes
+        if (quotesBiggerThanAmount.size() == 0) {
+            BigDecimal minValue = quotes.get(0).getVolumeSize();
+            int minValueIndex = 0;
+            for (int i = 0; i < quotes.size(); i++) {
+                if (quotes.get(i).getVolumeSize().compareTo(minValue) < 0
+                        && quotes.get(i).getVolumeSize().compareTo(BigDecimal.valueOf(0)) >0 ) {
```

Comment:
WhitespaceAround: &#39;&gt;&#39; is not followed by whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a146635345>*